src/inspector/tracing_agent.cc

```diff
@@ -144,23 +143,15 @@ DispatchResponse TracingAgent::start(
         "Tracing properties can only be changed through main thread sessions");
   }
 
-  std::set<std::string> categories_set;
-  protocol::Array<std::string>* categories =
-      traceConfig->getIncludedCategories();
-  for (size_t i = 0; i < categories->length(); i++)
-    categories_set.insert(categories->get(i));
-
-  if (categories_set.empty())
-    return DispatchResponse::Error("At least one category should be enabled");
+  auto categories = traceConfig->getIncludedCategories();
+  if (categories->length() == 0)
+    return DispatchResponse::Error("At least one category must be enabled");
 
   tracing::AgentWriterHandle* writer = GetTracingAgentWriter();
-  if (writer != nullptr) {
-    trace_writer_ =
-        writer->agent()->AddClient(categories_set,
-                                   std::make_unique<InspectorTraceWriter>(
-                                       frontend_object_id_, main_thread_),
-                                   tracing::Agent::kIgnoreDefaultCategories);
-  }
+  if (writer)
+    trace_writer_ = writer->agent()->AddClient({categories->begin(), categories->end()},
+                                                std::make_unique<InspectorTraceWriter>(frontend_object_id_, main_thread_),
+                                                tracing::Agent::kIgnoreDefaultCategories);
```

Comment:
@LordLucass ... I believe this is what @addaleax has in mind. Specifically, multiline if blocks need to use the surrounding `{ ... }` and c++ file line-lengths should not exceed 80 chars.

```suggestion
  if (writer != nullptr) {
    trace_writer_ = writer->agent()->AddClient({
      categories->begin(),
      categories->end()},
      std::make_unique<InspectorTraceWriter>(frontend_object_id_, main_thread_),
      tracing::Agent::kIgnoreDefaultCategories);
  }
```

Reply:
I understand. Thanks for the explanation, it cleared up my doubts.